python/paddle/fluid/tests/unittests/test_elementwise_sub_op.py

```diff
@@ -289,6 +289,32 @@ def _executed_api(self):
         self.subtract = paddle.subtract_
 
 
+class TestSubtractInplaceBroadcast(unittest.TestCase):
+    def test_broadcast_success(self):
+        paddle.disable_static()
+        x_numpy = np.random.rand(2, 3, 4).astype('float')
+        y_numpy = np.random.rand(3, 4).astype('float')
+
+        x = paddle.to_tensor(x_numpy)
+        y = paddle.to_tensor(y_numpy)
+
+        inplace_result = paddle.subtract_(x, y)
+        numpy_result = x_numpy - y_numpy
+        self.assertEqual((inplace_result.numpy() == numpy_result).all(), True)
+        paddle.enable_static()
+
+    def test_broadcast_errors(self):
+        paddle.disable_static()
+        x = paddle.rand([3, 4])
+        y = paddle.rand([2, 3, 4])
```

Comment:
add more comlecated broadcast testcases, e.g., [1,2,1,3] and [4,2,5,1]

Reply:
done